docs/source/java/examples/quickstartguide.rst

```diff
@@ -0,0 +1,277 @@
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: java
+.. highlight:: java
+
+=================
+Quick Start Guide
+=================
+
+.. contents::
+
+Arrow Java provides several building blocks; ``data types`` describe the types of values;
+``ValueVectors`` are sequences of typed values; ``fields`` describe the types of columns in
+tabular data; ``schemas`` describe a sequence of columns in tabular data and
+``VectorSchemaRoot`` represents tabular data. Arrow also provides ``readers`` and
+``writers`` for loading data from and persisting data to storage.
+
+Create A Value Vector
+*********************
+
+Also known as "arrays" in the columnar format. Represent a one-dimensional
+sequence of homogeneous values.
+
+**Int Vector**: Create an value vector of int32s like this [1, null, 2]
+
+.. code-block:: Java
+
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.IntVector;
+
+    try(BufferAllocator rootAllocator = new RootAllocator();
+        IntVector intVector = new IntVector("fixed-size-primitive-layout", rootAllocator)){
+        intVector.allocateNew(3);
+        intVector.set(0,1);
+        intVector.setNull(1);
+        intVector.set(2,2);
+        intVector.setValueCount(3);
+        System.out.println("Vector created in memory: " + intVector);
+    }
+
+**Varchar Vector**: Create an value vector of string like this [one, two, three]
+
+.. code-block:: Java
+
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.VarCharVector;
+
+    try(BufferAllocator rootAllocator = new RootAllocator();
+        VarCharVector varCharVector = new VarCharVector("variable-size-primitive-layout", rootAllocator)){
+        varCharVector.allocateNew(3);
+        varCharVector.set(0, "one".getBytes());
+        varCharVector.set(1, "two".getBytes());
+        varCharVector.set(2, "three".getBytes());
+        varCharVector.setValueCount(3);
+        System.out.println("Vector created in memory: " + varCharVector);
+    }
+
+Create A Field
+**************
+
+Fields are used to denote the particular columns of tabular data.
+
+**Field**: Create a column "document" of string type with metadata.
+
+.. code-block:: Java
+
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+
+    Map<String, String> metadata = new HashMap<>();
+    metadata.put("A", "Id card");
+    metadata.put("B", "Passport");
+    metadata.put("C", "Visa");
+    Field document = new Field("document", new FieldType(true, new ArrowType.Utf8(), /*dictionary*/ null, metadata), /*children*/ null);
```

Comment:
Can we break long lines to be more readable?

Reply:
Changed